templates/statefulset.yaml

```diff
@@ -27,6 +27,12 @@ spec:
         clusterName: {{ include "infinispan-helm-charts.name" . }}
         {{- include "infinispan-helm-charts.podLabels" . | nindent 8 }}
     spec:
+      {{- if .Values.nodeSelector }}
```

Comment:
Currently the `nodeSelector` and `nodeAffinity` values set in the Values.yaml are not being reflected in the generated StatefulSet. You need to update  `Values.nodeSelector` and `Values.nodeAffinity`, to `Values.deploy.nodeSelector` and `Values.deploy.nodeAffinity`, respectively.